code/modules/overmap/objects/event_datum.dm

```diff
@@ -78,151 +130,312 @@
+	chain_rate = 1
+	strength = 2
+	chance_to_affect = 15
+	interference_power = 15
+
+/datum/overmap/event/flare/major
+	name = "solar flare (major)"
+	base_icon_state = "flare_major_"
+	chance_to_affect = 60
+	interference_power = 60
+	chain_rate = 4
+	strength = 8
+
 ///ELECTRICAL STORM - explodes your computer and IPCs
 /datum/overmap/event/electric
 	name = "electrical storm (moderate)"
-	desc = "A spatial anomaly, an unfortunately common sight on the frontier. Disturbing it tends to lead to intense electrical discharges"
-	token_icon_state = "electrical_moderate1"
+	desc = "A buildup of static electrity, an unfortunately common sight on the frontier. Disturbing it tends to lead to intense electrical discharges"
+	base_icon_state = "electrical_medium_"
+	default_color = "#e8e85c"
 	chance_to_affect = 15
 	spread_chance = 30
 	chain_rate = 3
+	interference_power = 15
 	var/zap_flag = ZAP_STORM_FLAGS
-	var/max_damage = 15
-	var/min_damage = 5
+	var/max_damage = 3000
+	var/min_damage = 1000
 
-/datum/overmap/event/electric/Initialize(position, ...)
-	. = ..()
-	switch(type) //woop! this picks one of two icon states for the severity of the storm in overmap.dmi
-		if(/datum/overmap/event/electric/minor)
-			token.icon_state = "electrical_minor[rand(1, 2)]"
-		if(/datum/overmap/event/electric)
-			token.icon_state = "electrical_moderate[rand(1, 2)]"
-		if(/datum/overmap/event/electric/major)
-			token.icon_state = "electrical_major[rand(1, 2)]"
-		else
-			token.icon_state = "electrical_moderate1"
-
-	token.color = "#e8e85c"
-	token.light_color = "#e8e85c"
-	token.update_appearance()
+/datum/overmap/event/electric/alter_token_appearance()
+	icon_suffix = "[rand(1, 4)]"
+	..()
+	if(current_overmap.override_object_colors)
+		token.color = current_overmap.hazard_primary_color
+	current_overmap.post_edit_token_state(src)
 
 /datum/overmap/event/electric/affect_ship(datum/overmap/ship/controlled/S)
 	var/datum/virtual_level/ship_vlevel = S.shuttle_port.get_virtual_level()
 	var/turf/source = ship_vlevel.get_side_turf(pick(GLOB.cardinals))
-	tesla_zap(source, 10, TESLA_DEFAULT_POWER, zap_flag)
-	for(var/mob/M as anything in GLOB.player_list)
-		if(S.shuttle_port.is_in_shuttle_bounds(M))
-			M.playsound_local(source, 'sound/magic/lightningshock.ogg', rand(min_damage / 10, max_damage / 10))
+	tesla_zap(source, 32, rand(min_damage, max_damage), zap_flag)
+
+	for(var/mob/poor_crew as anything in GLOB.player_list)
+		if(S.shuttle_port.is_in_shuttle_bounds(poor_crew))
+			poor_crew.playsound_local(poor_crew, THUNDER_SOUND, rand(min_damage, max_damage))
+
+
+/datum/overmap/event/electric/modify_emptyspace_mapgen(datum/overmap/dynamic/our_planet)
+	our_planet.weather_controller_type = /datum/weather_controller/shrouded
+	return ..()
 
 /datum/overmap/event/electric/minor
 	name = "electrical storm (minor)"
+	base_icon_state = "electrical_minor_"
+	interference_power = 5
 	spread_chance = 40
 	chain_rate = 2
-	max_damage = 10
-	min_damage = 3
+	max_damage = 1000
```

Comment:
These usually won't do anything afaik since telsa zaps stop once their power is less than 1000